Main/Wordle/Wordle/WordleGame.cpp

```diff
@@ -0,0 +1,190 @@
+#include <iostream>
+#include <fstream>
+#include <ctime>
+#include <unordered_set>
+#include <algorithm>
+
+constexpr uint_fast8_t WORD_LENGTH = 5;
+
+WordleGame::WordleGame(const std::string& filePath) : attempts(0), wordOfTheDayMode(false) {
+    loadDatabase(filePath);
+    srand(static_cast<unsigned int>(time(nullptr)));
+}
+
+void WordleGame::start() {
+    int choice;
+    while (true) {
+        displayMenu();
+        std::cin >> choice;
+
+        if (choice == 0) {
+            break;
+        }
+        else if (choice == 1) {
+            if (isWordOfTheDayGuessed()) {
+                std::cout << "Already found! Come back tomorrow!\n";
+                continue;
+            }
+            targetWord = getWordOfTheDay();
+            wordOfTheDayMode = true;
+        }
+        else if (choice == 2) {
+            targetWord = getRandomWord();
+            wordOfTheDayMode = false;
+        }
+        else {
+            std::cout << "Invalid choice. Try again.\n";
+            continue;
+        }
+
+        resetGame();
+        play();
+    }
+}
+
+uint_fast16_t WordleGame::getCurrentDate() {
+    constexpr uint_fast8_t  SECONDS_IN_MINUTE = 60;
+    constexpr uint_fast8_t  MINUTES_IN_HOUR = 60;
+    constexpr uint_fast8_t  HOURS_IN_DAY = 24;
+    constexpr uint_fast32_t SECONDS_IN_DAY = SECONDS_IN_MINUTE * MINUTES_IN_HOUR * HOURS_IN_DAY;
+    std::time(nullptr);
+    const time_t now = time(nullptr);
+
+   
+    tm localTime{};
+    localtime_s( &localTime, &now);
+
+    return mktime(&localTime) / SECONDS_IN_DAY;
+}
+
+std::string WordleGame::getRandomWord() {
+    const size_t randomIndex = rand() % database.size();
+    return database[randomIndex];
+}
+
+std::string WordleGame::getWordOfTheDay() {
+    const uint_fast16_t todayIndex = WordleGame::getCurrentDate() % database.size();
+    return database[todayIndex];
+}
+
+std::string WordleGame::compareWords(const std::string& guess) {
+    std::string                   result = "*****";
+    std::string                   tempTarget = targetWord;
+    std::unordered_multiset<char> unmatchedTargetChars;
+
+    for (uint_fast8_t i = 0; i < WORD_LENGTH; ++i) {
+        if (guess[i] == targetWord[i]) {
+            result[i] = toupper(guess[i]);
+            tempTarget[i] = '*';
+        }
+        else {
+            unmatchedTargetChars.insert(tempTarget[i]);
+        }
+    }
+
+    for (uint_fast8_t i = 0; i < WORD_LENGTH; ++i) {
+        if (result[i] == '*') {
+            auto it = unmatchedTargetChars.find(guess[i]);
+            if (it != unmatchedTargetChars.end()) {
+                result[i] = tolower(guess[i]);
+                unmatchedTargetChars.erase(it);
+            }
+        }
+    }
+
+    return result;
+}
+
+bool WordleGame::isWordOfTheDayGuessed() {
+    std::ifstream file("word_of_the_day_status.txt", std::ios::binary);
+    if (file.is_open()) {
+        uint_fast16_t date;
+        file.read(reinterpret_cast<char*>(&date), sizeof(date));
```

Comment:
prefer going with no `reintepret_cast` solution. It's very unsafe ad error-prone

Reply:
reading date into a string and then parsing that string into int format would be safer even though slower and more code would be needed